internal/tasks/manifests.go

```diff
@@ -711,12 +711,14 @@ func (j *Janitor) doVulnerabilityCheck(account keppel.Account, repo keppel.Repos
 	//ask Clair for vulnerability status of blobs in this image
 	vulnInfo.Message = "" //unless it gets set to something else below
 	if len(layerBlobs) > 0 {
-		clairState, err := j.cfg.ClairClient.CheckManifestState(manifest.Digest, func() (clair.Manifest, error) {
+		clairState, err := j.cfg.ClairClient.CheckManifestState(manifest.Digest, j.timeNow, func() (clair.Manifest, error) {
 			return j.buildClairManifest(account, manifest, layerBlobs)
 		})
 		if err != nil {
 			return err
 		}
+		vulnInfo.IndexStartedAt = clairState.IndexingStartedAt
+		vulnInfo.IndexFinishedAt = clairState.IndexingFinishedAt
```

Comment:
After now reviewing the relevant code in its entirety, I don't think we need to change `CheckManifestState` at all. Please correct me if I'm wrong, but as far as I can see:

- `vulnInfo.IndexStartedAt` needs to be filled here if it is currently nil **or** `clairState.IndexingWasRestarted`.
- `vulnInfo.IndexFinishedAt` needs to be filled here if it is currently nil **and** `clairState.IsIndexed`.

Reply:
🤔